From 5551d21b5d1d6aa3e266ef8b1cf65bc6725e1171 Mon Sep 17 00:00:00 2001
From: Raito Bezarius <masterancpp@gmail.com>
Date: Wed, 1 Feb 2023 04:12:52 +0100
Subject: [PATCH] crypto: remove salsa20 as they are not here anymore

---
 crypto/Kconfig  | 12 ------------
 crypto/Makefile |  1 -
 2 files changed, 13 deletions(-)

diff --git a/crypto/Kconfig b/crypto/Kconfig
index 0239e513ea5a..5e2890ca74ce 100644
--- a/crypto/Kconfig
+++ b/crypto/Kconfig
@@ -1404,18 +1404,6 @@ config CRYPTO_KHAZAD
 	  See also:
 	  <http://www.larc.usp.br/~pbarreto/KhazadPage.html>
 
-config CRYPTO_SALSA20
-	tristate "Salsa20 stream cipher algorithm"
-	select CRYPTO_SKCIPHER
-	help
-	  Salsa20 stream cipher algorithm.
-
-	  Salsa20 is a stream cipher submitted to eSTREAM, the ECRYPT
-	  Stream Cipher Project. See <https://www.ecrypt.eu.org/stream/>
-
-	  The Salsa20 stream cipher algorithm is designed by Daniel J.
-	  Bernstein <djb@cr.yp.to>. See <https://cr.yp.to/snuffle.html>
-
 config CRYPTO_CHACHA20
 	tristate "ChaCha stream cipher algorithms"
 	select CRYPTO_LIB_CHACHA_GENERIC
diff --git a/crypto/Makefile b/crypto/Makefile
index ccbf35cd42c2..624fb42bbf2c 100644
--- a/crypto/Makefile
+++ b/crypto/Makefile
@@ -142,7 +142,6 @@ obj-$(CONFIG_CRYPTO_TEA) += tea.o
 obj-$(CONFIG_CRYPTO_KHAZAD) += khazad.o
 obj-$(CONFIG_CRYPTO_ANUBIS) += anubis.o
 obj-$(CONFIG_CRYPTO_SEED) += seed.o
-obj-$(CONFIG_CRYPTO_SALSA20) += salsa20_generic.o
 obj-$(CONFIG_CRYPTO_CHACHA20) += chacha_generic.o
 obj-$(CONFIG_CRYPTO_POLY1305) += poly1305_generic.o
 obj-$(CONFIG_CRYPTO_DEFLATE) += deflate.o
-- 
2.39.0

